examples/bls12-381/src/bls12-381.rs

```diff
@@ -44,14 +44,14 @@ pub type Fp12 = (Fp6, Fp6); //w² = v
 
 
 /* Arithmetic for FP2 elements */
-fn fp2zero() -> Fp2 {
-    fp2fromfp(Fp::ZERO())
-}
-
 fn fp2fromfp(n: Fp) -> Fp2 {
     (n, Fp::ZERO())
 }
 
+fn fp2zero() -> Fp2 {
```

Comment:
Is this change necessary for the Coq compiler to work or is this an accidental/unrelated change.
I'm just asking because if this is necessary it looks like a compiler or language issue we should fix.

Reply:
Yes, this is necessary at the moment because the order of definitions are preserved from rust to rustspec, but Coq (and the other backends?) does not allow backwards references. This is actually also an issue in some of the other examples.

Reply:
@mikkelmilo is right, Rust has an advanced out-of-order handling whereas ML-based languages like Coq or F* require the source code to be ordered by forward dependency. Inside the compiler, I replicate Rust's semantic for that but I could switch it to a stricter semantics where you can only use what you have defined above.

Reply:
Ok, this is a good point. I think people coming from C wouldn't mind a stricter ordering requirement as that's what they are used to. For everyone coming from Rust this is unexpected though. But then the out-of-order handling doesn't work for everything in Rust (in particular macros must be defined before use). So maybe we should enforce the order in hacspec.